explorer/graphql/data/provider_helpers.go

```diff
@@ -0,0 +1,121 @@
+package data
+
+import (
+	"github.com/omni-network/omni/explorer/db/ent"
+	"github.com/omni-network/omni/explorer/graphql/resolvers"
+	"github.com/omni-network/omni/lib/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/graph-gophers/graphql-go"
+)
+
+// EntBlockToGraphQLBlock converts an ent.Block to a resolvers.XBlock.
+func EntBlockToGraphQLBlock(block *ent.Block) (*resolvers.XBlock, error) {
+	sourceChainIDBig, err := hexutil.DecodeBig(hexutil.EncodeUint64(block.SourceChainID))
```

Comment:
We invoke this `hexutil.DecodeBig(hexutil.EncodeUint64(...))` 15 times. Maybe add a little helper func to the `data` package, e.g. `uint2Big`.